lib/api/multiObjectDelete.js

```diff
@@ -0,0 +1,307 @@
+        function checkPolicies(quietSetting, objects, next) {
+            // if request from account, no need to check policies
+            if (!authInfo.isRequesterAnIAMUser()) {
+                inPlay = objects.slice();
+                return next(null, quietSetting);
+            }
+
+            // TODO: once arsenal's extractParams is separated from doAuth
+            // function, refactor so only extract once and send
+            // params on to this api
+            const authParams = auth.server.extractParams(request, log,
+                's3', request.query);
+            const requestContextParams = {
+                constantParams: {
+                    headers: request.headers,
+                    query: request.query,
+                    generalResource: request.bucketName,
+                    requesterIp: request.socket.remoteAddress,
+                    sslEnabled: request.connection.encrypted,
+                    apiMethod: 'objectDelete',
+                    awsService: 's3',
+                    locationConstraint: null,
+                    requesterInfo: authInfo,
+                    signatureVersion: authParams.params.data.authType,
+                    authType: authParams.params.data.signatureVersion,
+                    signatureAge: authParams.params.data.signatureAge,
+                },
+                parameterize: {
+                    specificResource: objects,
+                },
+            };
+            return vault.checkPolicies(requestContextParams, authInfo.getArn(),
+                log, (err, authorizationResults) => {
+                    // there were no policies so received a blanket AccessDenied
+                    if (err && err.AccessDenied) {
+                        objects.forEach(key => {
+                            errorResults.push({
+                                key,
+                                error: errors.AccessDenied });
+                        });
+                        return next(null, quietSetting);
+                    }
+                    if (err) {
+                        log.trace('error checking policies', { error: err });
+                        return next(err);
+                    }
+                    if (objects.length !== authorizationResults.length) {
+                        log.error('vault did not return correct number of ' +
+                        'authorization results');
+                        return next(errors.InternalError);
+                    }
+                    for (let i = 0; i < authorizationResults.length; i++) {
+                        const result = authorizationResults[i];
+                        // result is { isAllowed: true,
+                        // arn: arn:aws:s3:::bucket/object} unless not allowed
+                        // in which case no isAllowed key will be present
+                        const slashIndex = result.arn.indexOf('/');
+                        if (slashIndex === undefined) {
```

Comment:
`indexOf()` returns -1 in case it doesn't find the string, not `undefined`. The fact that this is not picked up by the tests means that this case is not handled in unit testing.